Ghidra/Debug/Debugger-agent-dbgmodel/src/main/java/agent/dbgmodel/model/impl/DbgModel2TargetObjectImpl.java

```diff
@@ -248,7 +239,7 @@ protected CompletableFuture<Void> addModelObjectAttributes(Map<String, Object> a
 			if (proxy instanceof TargetAccessConditioned) {
 				attrs.put(TargetAccessConditioned.ACCESSIBLE_ATTRIBUTE_NAME, accessible);
 			}
-			if (proxy instanceof TargetExecutionStateful) {
+			else if (proxy instanceof TargetExecutionStateful) {
```

Comment:
These interfaces are not mutually exclusive. This model can dynamically construct new classes (see ProxyUtilities.composeOnDelegate) which implement many TargetX interfaces. This logic ensures that such an object also includes all the attributes required by its respective interfaces.